hello-world-react-app/src/components/CounterFunction.tsx

```diff
@@ -0,0 +1,16 @@
+import { useState } from "react";
+
+export const CounterFunction = () => {
+  const [count, setCount] = useState(0);
+
+  return (
+    <>
+      <h1>CounterFunction</h1>
+      {count}
+      <div>
+        <button onClick={() => setCount(count + 1)}>increment</button>
```

Comment:
for learning purpose it's ok, but in production - I think it's better to do some method like handleClick and do logic there. Cause you in future you may need to add some logic to it, for example - analytics

Reply:
yep, totally agree.
I moved handlers to separate functions in other implementations